### PR TITLE
Add manager_factory to create data_manager

### PR DESCRIPTION
**Pull request by junjiejiangjjj**

### CODE REVIEW

File: gptcache/manager/factory.py [hunk 1]
```diff
@@ -1,8 +1,91 @@
...
+    if scalar_params is None:
+        scalar_params = {}
+    if scalar == "sqlite":
+        scalar_params["sql_url"] = "sqlite:///" + os.path.join(data_dir, "sqlite.db")
+    s = CacheBase(name=scalar, **scalar_params)
+
+    if vector_params is None:
+        vector_params = {}
+    if vector == "faiss":
+        vector_params["index_path"] = os.path.join(data_dir, "faiss.bin")
```
Comment (SimFG) [inline gptcache/manager/factory.py:74]: the default is `faiss.index`, if you want to change it, please change other relative codes

File: gptcache/manager/factory.py [hunk 2]
```diff
@@ -1,8 +1,91 @@
...
+    elif vector == "hnswlib":
+        vector_params["index_path"] = os.path.join(data_dir, "hnswlib_index.bin")
```
Comment (SimFG) [inline gptcache/manager/factory.py:76]: please use the `.index`

File: gptcache/manager/factory.py (hunk 1 above)
Comment (junjiejiangjjj) [reply gptcache/manager/factory.py]: fixed

File: gptcache/manager/factory.py (hunk 2 above)
Comment (junjiejiangjjj) [reply gptcache/manager/factory.py]: fixed

**Pull request merged**